userscripts/user_exp_time_matrix.js

```diff
@@ -0,0 +1,453 @@
+            return false;
+        }
+
+        return true;
+    };
+
+    this.makeModal = function (event) {
+        var namePrefix = this.config.extensionNamePrefix;
+        var out = "";
+
+        out += '<div id="' + namePrefix + 'Result"></div>';
+
+        return out;
+    };
+
+    this.setModalHeaderFooter = function () {
+        var _this = this;
+        var modalName = this.config.modal.name;
+        var namePrefix = this.config.extensionNamePrefix;
+        var elements = this.config.elements;
+        var treasures = this.config.specs.treasures;
+        var adventures = this.config.specs.adventures;
+
+        var headerRow1 = createTableRow([
+          [3, createSwitch(
+                elements.PFB_RADIO,
+                ExplorerTimeMatrix_PFB_RB_STATUS
+              ) +
+                '<div style="position: absolute;left: 55px;top: 1px;" id="' +
+                elements.PFB_RADIO_TEXT +
+                '">{0}</div>'.format(
+                  ExplorerTimeMatrix_PFB_RB_STATUS ? _this.config.texts.SHOW_PFB_TIMES : _this.config.texts.SHOW_NORMAL_TIMES
+                )],
+          [3, createSwitch(
+            elements.ADV_RADIO,
+            ExplorerTimeMatrix_ADV_RB_STATUS
+          ) +
+            '<div style="position: absolute;left: 55px;top: 1px;" id="' +
+            elements.ADV_RADIO_TEXT +
+            '">{0}</div>'.format(
+              ExplorerTimeMatrix_ADV_RB_STATUS ? _this.config.texts.FIND_ADVENTURE : _this.config.texts.FIND_TREASURE
+            )],
+          [6, '']
+        ], true);
+
+        // treasure
+        var _treasure = '<table style="width: 100%;"><tbody><tr>';
+        treasures.forEach(function(item) {
+          _treasure += '<td style="width: 14.2%; color:#fff; font-size: 11px;">' + getImageTag(item.icon, '25px') + '</td>';
+        });
+        _treasure += '</tr></tbody></table>';
+
+        // adventure
+        var _adventure = '<table style="width: 100%;"><tbody><tr>';
+        adventures.forEach(function(item) {
+          _adventure += '<td style="width: 25%; color:#fff; font-size: 11px;">' + getImageTag(item.icon, '25px') + '</td>';
+        });
+        _adventure += '</tr></tbody></table>';
+
+        var headerRow2 = createTableRow([
+                [3, loca.GetText("LAB","Name")],
+                [9, _treasure, elements.TREASURE_TIMES],
+                [9, _adventure, elements.ADVENTURE_TIMES]
+          ], true);
+
+        $("#" + modalName + " .modal-header").append([$('<br/>'), $('<div>', {'class': 'depletedInfo text-center'}), $('<div>', {'class': 'container-fluid'}).html(headerRow1 + headerRow2)]);
+    };
+
+    this._isPFBActive = function() {
+      game.gi.mZoneBuffManager.isBuffRunning("MultiplierBuffZone2_PremiumFriendBuff1Day")
+    };
+
+    this.getAvailableExplorers = function () {
+      var _this = this;
+      var availableExplorers = [];
+
+      game
+        .getSpecialists()
+        .sort(0)
```

Comment:
players can choose sort method in settings.. try to use sort(specNameSorter)
specNameSorter is function from common.js